python/ray/tune/ray_trial_executor.py

```diff
@@ -198,6 +216,15 @@ def stage_and_update_status(self, trials: List[Trial]):
 
         Stages placement groups of all trials.
         """
+        if not self._has_cleaned_up_pgs:
+            # Clean up existing placement groups after trigger the tuning
+            # run step() method for the first time
+            self._pg_manager.cleanup_existing_pg()
+            self._has_cleaned_up_pgs = True
```

Comment:
hmm, what does this do?

Reply:
It cleans up existing placement groups at the start of the trial. If the tune run is exited ungracefully, placement groups are still around. If we then start a new run placement groups cannot be scheduled because the resources are occupied by the old trial.

In that case, cleaning up after the run does not really suffice because we never reach that part if the driver errors or is aborted.